Add LookupSelf capability to Auth

EXPEddrewery opened the pull request:
Under some circumstances, it's necessary to lookup a token to ensure that it is still valid.

Currently the library does not have any facility to do this.

I have created a separate response as it is significantly different to the other Auth responses, particular in the use of "data" as opposed to "auth" in the response.

steve-perkins commented:
It looks like you created a new response type, `LookupResponse`, but there is no such class in this PR or in your own fork.  Did you forget to commit that file, maybe?

EXPEddrewery commented:
My apologies @steve-perkins! I thought I pushed everything but it seems it was ignored due to the "com/" in the .gitignore file.

I've added it now and added a couple of PMD warning suppressions that popped up for the lookupSelf method.

steve-perkins commented:
Thanks so much!

EXPEddrewery commented:
No worries.
